### ISSUE TITLE
Add `pulumi project refresh`

### ISSUE THREAD

**Issue by EvanBoyle** [pulumi/pulumi#11183]
I'm writing some programs that do automated refresh and destroy operations over a bunch of stacks. I don't have the source code for these stacks, just the fully qualified stack names. My process is:

1. construct a dummy `pulumi.yaml` file with the correct project name and dummy runtime info
2. write that yaml into a directory
3. run `pulumi stack select foo/bar/buzz && pulumi config refresh`
4. run my refresh or destroy operation

I'd love to instead turn this into:

1. `pulumi project refresh foo/bar`
2. `pulumi stack select foo/bar/buzz && pulumi config refresh`

Now I don't have to manipulate and construct YAML for this scenario. This problem becomes more acute with the introduction of project-level config https://github.com/pulumi/pulumi/issues/2307 as now reconstructing a dummy project file will not include any config and might lead to correctness issues.

**Comment by iwahbe** [comment]
I don't think that project level config will break your workflow, since all inputs are still recorded in the state, regardless of which config source they originated from. It's a good idea to have in general though.

**Comment by EvanBoyle** [comment]
Another pretty cool part about this for yaml specifically is that Pulumi would store your last applied YAML program.
